Source/WarpX.cpp

```diff
@@ -790,31 +790,37 @@ WarpX::ReadParameters ()
         queryWithParser(pp_algo, "costs_heuristic_cells_wt", costs_heuristic_cells_wt);
         queryWithParser(pp_algo, "costs_heuristic_particles_wt", costs_heuristic_particles_wt);
 
-        int particle_shape;
-        if (pp_algo.query("particle_shape", particle_shape) == false)
-        {
-            amrex::Abort("\nalgo.particle_shape must be set in the input file:"
-                         "\nplease set algo.particle_shape to 1, 2, or 3");
-        }
-        else
-        {
-            if (particle_shape < 1 || particle_shape > 3)
+        // check if any nspecies > 0 before testing entry for algo.particle_shape
```

Comment:
Sorry for the very minor detail, I can commit the small change directly, to speed up the merge. Just want to make sure that the inline comment is as clear as possible (we are both checking and parsing the entry for `algo.particle_shape` at the same time, so this is also the place where we possibly assign the particle shape order and not only test that the user entered something acceptable):
```suggestion
        // Parse algo.particle_shape and check that input is acceptable
        // (do this only if there is at least one particle species)
```

Reply:
thank you @EZoni